### DISCUSSION TITLE
Change Content-Security-Policy header from plugin

### DISCUSSION THREAD

**Discussion by Findus23** [pretalx/pretalx#1143] [Q&A] [answered]
Hello everyone,

I am trying to add [Matomo](https://matomo.org) to a pretalx site and with the `pretalx.cfp.signals.html_head` signal it is easy to add the Javascript. But in addition I need to update the CSP header to also allow loading Javascript from this separate host.

What is the easiest way to accomplish this (either in the plugin or in pretalx itself).

Editing 
https://github.com/pretalx/pretalx/blob/f2e0ceebd1ce6553ffecc795c994c9e250b40e1d/src/pretalx/settings.py#L133
would work, but doesn't seem ideal to me.

**Comment by rixx** [comment, marked as answer]
Since this is the first request for this (and I'm lazy), you're getting a slightly different solution for now: I added a configuration option to pretalx to expand the CSP headers. Added and documented here: 0bbc718e

**Reply by Findus23** [reply]
That's an even better solution. In the meantime I solved it for my site by changing the values in settings.py directly.
